src/authgss_hash.c

```diff
@@ -70,8 +70,9 @@ static struct authgss_hash_st authgss_hash_st = {
 static inline uint64_t
 gss_ctx_hash(gss_union_ctx_id_desc *gss_ctx)
 {
-	return ((uint64_t) gss_ctx->mech_type +
-		(uint64_t) gss_ctx->internal_ctx_id);
+	/* mash together two pointers, could be more sophisticated? */
+	return ((uint64_t)((uintptr_t)gss_ctx->mech_type +
+			   (uintptr_t)gss_ctx->internal_ctx_id));
```

Comment:
I believe, for the math to be 64-bit on a 32-bit system, the cast needs to be:
(uint64_t)(uintptr_t)gss_ctx->mech_type + (uint64_t)(uintptr_t)gss_ctx->internal_ctx_id

Otherwise, the + op is done on uintptr_t (32-bit) and then promoted to uint64_t (final cast) causing potential overflow.